cedar-lean/Cedar/Thm/Data/List.lean

```diff
@@ -841,73 +915,323 @@ theorem forall₂_implies_all_right {α β} {R : α → β → Prop} {xs : List
       exists x
       simp only [mem_cons, ih, or_true, and_self]
 
-/-! ### mapM' -/
+/-! ### mapM, mapM', and mapM₁ -/
 
-theorem mapM'_ok_implies_forall₂ {α β γ} {f : α → Except γ β} {xs : List α} {ys : List β} :
-  List.mapM' f xs = .ok ys →
```

Comment:
I needed the converse as well, so I made this an iff and renamed it `mapM'_ok_iff_forall2`.  This lemma was introduced in #297.  I reformulated the proofs in #291 (and here) to rely on this lemma instead of the other lemmas in #291.